src/metabase/lib/js.cljs

```diff
@@ -982,6 +982,29 @@
   [a-query stage-number]
   (to-array (lib.core/filters a-query stage-number)))
 
+;; ## Dashboard filtering
+;; This is a special case of filtering. While linking dashboard filters to a question, there's no way to choose the
+;; stage of interest. Currently we default to the last stage, but this is a bit clumsy if there's eg. a custom column
+;; added after an aggregation. It's much more likely that the interesting thing to filter is the *last stage with
+;; aggregations*. That is, filtering the *inputs* to aggregations rather than their *output*.
+
+;; Filtering the post-aggregation values is a legitimate thing to do, and there are plans to surface both of those
+;; sets of columns. But that requires UI changes, while we can improve the UX significantly by being smart about the
+;; stage number we return here.
+(defn ^:export dashboard-filter-stage-index
```

Comment:
This should be tested, and it seems that the implementation should be in fe-utils.